torch/csrc/autograd/python_variable.cpp

```diff
@@ -303,9 +346,12 @@ static PyObject* THPVariable_as_subclass(PyObject* _self, PyObject* args, PyObje
   if (!PyType_Check(cls)) {
     throw torch::TypeError("cls must be a type (got %s)", Py_TYPE(cls)->tp_name);
   }
+  auto data = self.alias();
+  unsafe_clear_pyobj(data);
```

Comment:
Yeah. So instead of clearing the PyObject, I'd rather just error here (preferably because we attempted to do a `__torch_function__` dispatch and no one said they knew how to do this subclass conversion

Reply:
But what about subclasses like `nn.Parameter` for which `__torch_function__` is disabled?